arangod/GeneralServer/VstCommTask.cpp

```diff
@@ -266,13 +266,6 @@ void VstCommTask<T>::processMessage(velocypack::Buffer<uint8_t> buffer, uint64_t
   // handle request types
   if (mt == fu::MessageType::Authentication) {  // auth
     handleVstAuthRequest(VPackSlice(buffer.data()), messageId);
-    // Separate superuser traffic:
-    // Note that currently, velocystream traffic will never come from
-    // a forwarding, since we always forward with HTTP.
-    if (_authMethod != AuthenticationMethod::NONE && _authenticated &&
-        _authToken.username().empty()) {
-      stat.SET_SUPERUSER();
-    }
```

Comment:
Does it mean that the VST authentication messages will never be counted as superuser traffic?
If so, is this a behavior change, and what's the rationale for it?
Thanks for elaborating!

Reply:
this code never worked the statistic object are removed once the auth gets a response and this is a heap access after free

Reply:
Ok, got it.
Looks to me like the move constructor of `RequestStatistics::Item` properly sets `_stat` to nullptr for the moved-from statistics object, and that all `SET_XXXX()` methods on it check if `_stat` is a nullptr before accessing it.
Anyway, in the best case the above code will have done nothing.
Does it make sense to move the code 
```
    if (_authMethod != AuthenticationMethod::NONE && _authenticated &&
        _authToken.username().empty()) {
      stat.SET_SUPERUSER();
    }
```
into the success case of `handleVstAuthRequest`, before `sendErrorResponse()` is called?

Reply:
Maybe _stat is not moved though it's just deleted. The stats handling is super messy

Reply:
True. Shall we move the `SET_SUPERUSER()` call into `handleVstAuthRequest()` then?
That should be safe.